pig/src/main/java/com/netflix/iceberg/pig/SchemaUtil.java

```diff
@@ -0,0 +1,144 @@
+ * limitations under the License.
+ */
+
+package com.netflix.iceberg.pig;
+
+import com.google.common.collect.Lists;
+import com.netflix.iceberg.Schema;
+import com.netflix.iceberg.types.Type;
+import com.netflix.iceberg.types.Types;
+import org.apache.pig.ResourceSchema;
+import org.apache.pig.ResourceSchema.ResourceFieldSchema;
+import org.apache.pig.data.DataType;
+import org.apache.pig.impl.logicalLayer.FrontendException;
+
+import java.io.IOException;
+import java.util.List;
+
+import static java.lang.String.format;
+import static com.netflix.iceberg.types.Types.*;
+
+public class SchemaUtil {
+
+  public static ResourceSchema convert(Schema icebergSchema) throws IOException {
+    ResourceSchema result = new ResourceSchema();
+    List<ResourceFieldSchema> fields = Lists.newArrayList();
+    for (Types.NestedField nf : icebergSchema.columns()) {
+      fields.add(convert(nf));
+    }
+    result.setFields(fields.toArray(new ResourceSchema.ResourceFieldSchema[0]));
+    return result;
+  }
+
+  private static ResourceFieldSchema convert(Types.NestedField field) throws IOException {
+    ResourceFieldSchema result = new ResourceFieldSchema();
+    result.setName(field.name());
+    result.setDescription(format("FieldId: %s", field.fieldId()));
+
+    result.setType(convertType(field.type()));
+
+    if (!field.type().isPrimitiveType()) {
+      result.setSchema(convertComplex(field));
+    }
+
+    return result;
+  }
+
+  private static ResourceFieldSchema convert(Type type) throws IOException {
+    ResourceFieldSchema result = new ResourceFieldSchema();
+    result.setType(convertType(type));
+
+    return result;
+  }
+
+  private static byte convertType(Type type) throws IOException {
+    switch (type.typeId()) {
+      case BOOLEAN:   return DataType.BOOLEAN;
+      case INTEGER:   return DataType.INTEGER;
+      case LONG:      return DataType.LONG;
+      case FLOAT:     return DataType.FLOAT;
+      case DOUBLE:    return DataType.DOUBLE;
+      case TIMESTAMP: return DataType.LONG;
+      case DATE:      return DataType.CHARARRAY;
+      case STRING:    return DataType.CHARARRAY;
+      case FIXED:     return DataType.BOOLEAN;
+      case BINARY:    return DataType.BYTEARRAY;
+      case DECIMAL:   return DataType.BIGDECIMAL;
+      case STRUCT:    return DataType.TUPLE;
+      case LIST:      return DataType.BAG;
+      case MAP:       return DataType.MAP;
+      default:
+        throw new FrontendException("Unsupported primitive type:" + type);
+    }
+  }
+
+  private static ResourceSchema convertComplex(Types.NestedField field) throws IOException {
+    ResourceSchema result = new ResourceSchema();
+
+    Type type = field.type();
+
+    switch (type.typeId()) {
+      case STRUCT:
+        StructType structType = type.asStructType();
+
+        List<ResourceFieldSchema> fields = Lists.newArrayList();
+
+        for (NestedField f : structType.fields()) {
+          fields.add(convert(f.type()));
```

Comment:
What happens to the field's name? Is there a way to create a `Tuple` with named fields?

Reply:
The project method uses findByName, which will return fields in nested structures. I think that's correct, but it doesn't appear to be tied to the type returned here. Are projections based entirely on a set of strings provided by the user and not Pig handing a Type to the loader?